storage-node/packages/runtime-api/balances.js

```diff
@@ -51,7 +51,7 @@ class BalancesApi {
    */
   async freeBalance(accountId) {
     const decoded = this.base.identities.keyring.decodeAddress(accountId, true)
-    return this.base.api.query.balances.freeBalance(decoded)
+    return (await this.base.api.derive.balances.all(decoded)).freeBalance
```

Comment:
In this context I think `(await this.base.api.derive.balances.all(decoded)).availableBalance` may actually be more suitable, because `freeBalance = avaialble + locked` and if the balance is locked it seems like you cannot transfer it or use it to pay the fee.

Reply:
Leszek's solution is only a single line. I'm including the fix in the current PR.